proposals/spline-animation/README.md

```diff
@@ -0,0 +1,1910 @@
+is possible (using very long tangents) to construct a spline segment that, under
+the Bezier rules, would cross over itself and form a loop.  Both Presto and Maya
+consider this an unacceptable condition, and excise the loop from the spline, so
+that every spline is indeed a function.
+
+Presto and Maya have different algorithms for this "functional forcing", and
+both have disadvantages.  Presto creates a vertical discontinuity at an
+arbitrary time.  Maya shortens tangents until the loop disappears, resulting in
+a smooth shape, but one that is very different than what is authored.
+
+We may wish to choose one of these algorithms, but a third would be more
+predictable: create a cusp where the Bezier curve overlaps itself.
+
+It is also theoretically possible to forbid tangents that cause crossovers.  But
+this would likely be surprising, and would violate our "All Splines Valid"
+principle (detailed below).
+
+## Dual-Valued Knots
+
+Value discontinuities may be introduced to a series by means of _dual-valued
+knots_.  A dual-valued knot has two values: an ordinary value, and a
+_pre-value_.  The value exactly at the knot time, and at any nonzero time delta
+after the knot, is the ordinary value.  The value at any nonzero time delta
+before the knot is the pre-value.  There is an instantaneous change in value at
+exactly the knot time.
+
+Any knot can be dual-valued, regardless of the interpolation methods of the
+adjacent segments.  However, if a dual-valued knot follows a held segment, the
+pre-value is ignored, because otherwise there would be a strange case where the
+pre-value was effective at only the left side of a single knot.  So in this
+case, if the pre-side value is queried, the value return is the held value from
+the prior knot.
+
+### Sided Queries
+
+The fundamental USD value resolution operation is `UsdAttribute::Get`.  Starting
+with USD Anim, clients may also call `GetPreValue`.  The two methods return the
+same value, unless the specified time is exactly at a dual-valued knot.
+
+Pre-values typically are not evaluated directly in order to render USD content.
+They exist as a mechanism for shaping curves.  Querying pre-values is often
+important for authoring systems, but usually not important for downstream
+evaluation.
+
+Sided queries will also affect other situations with value discontinuities:
+
+- At a knot at the end of a held segment.
+- At a time sample with held interpolation (e.g. for a string-valued attribute).
+- At the boundary of loop iterations in "reset" mode.
+- At "jump discontinuities" in value clips.
+
+See [Question: schema-level methods](#schema-level-methods).
+
+## Looping
+
+_Looping_ is the repeating of series regions.  Looping will be supported in two
+forms:
+
+- _Inner loops_ come from Presto.  These specify a _prototype region_, which is
+  repeated a finite number of times before and/or after the prototype region.
+  The repeated portions are called the _echo region_.  Inner loops use a mode
```

Comment:
I don't think I fully understand the difference between a prototype region and an echo region, since both seem to get repeated? Or is the echo region the part of the time line that includes the animation repetition, while the prototype region is the area that is being repeated? so an echo region is fed by a prototype region?

Reply:
Also would `source region` perhaps be a more accessible term than prototype region?